Modules/Loadable/Tables/Logic/vtkSlicerTablesLogic.cxx

```diff
@@ -68,27 +72,75 @@ ::AddTable(const char* fileName, const char* name)
     return 0;
     }
 
-  // Storage node
-  vtkNew<vtkMRMLTableStorageNode> tableStorageNode;
-  tableStorageNode->SetFileName(fileName);
-  this->GetMRMLScene()->AddNode(tableStorageNode.GetPointer());
-
   // Storable node
-  vtkNew<vtkMRMLTableNode> tableNode;
-  if (name)
-    {
-    tableNode->SetName(name);
-    }
-  this->GetMRMLScene()->AddNode(tableNode.GetPointer());
+  vtkMRMLTableNode *tableNode;
 
-  // Read
-  int res = tableStorageNode->ReadData(tableNode.GetPointer());
-  if (res == 0) // failed to read
+  if (std::string(fileName).find(".db") != std::string::npos)
     {
-    vtkErrorMacro("vtkSlicerTablesLogic::AddTable failed: failed to read data from file: "<<fileName);
-    this->GetMRMLScene()->RemoveNode(tableStorageNode.GetPointer());
-    this->GetMRMLScene()->RemoveNode(tableNode.GetPointer());
-    return 0;
+    // SQLite
+    std::string dbname = std::string("sqlite://") + std::string(fileName);
+    vtkSQLiteDatabase *database = vtkSQLiteDatabase::SafeDownCast( vtkSQLiteDatabase::CreateFromURL(dbname.c_str()));
+    if (!database->Open(name?name:"", vtkSQLiteDatabase::USE_EXISTING))
+      {
+      vtkErrorMacro("Failed to read tables from " << fileName);
+      database->Delete();
+      return 0;
+      }
+    vtkStringArray *tables = database->GetTables();
```

Comment:
user shouldbe able to specify a regexp (or some other filter) to choose which tables to read (and also specify a password)